ruth/data/map.py

```diff
@@ -45,7 +45,7 @@ def get_osm_segment_id(node_from: int, node_to: int):
     return f"OSM{node_from}T{node_to}"
 
 
-class BBox:
+class BBox(metaclass=Singleton):
```

Comment:
Why is this a singleton?

Reply:
There was issue that when the map creating is in the vehicle's post_init function, the bbox would register as different bbox for each car, so the Singleton on the map class was not working.
But it could probably be redone in a way that in the map constructor are 4 floats instead of the bbox or move the map from vehicle post_init alll together.

Reply:
Yeah, I think that we have brainstormed this before, getting rid of the map constructor would be really great =D

Reply:
I tried to move the Map into the Simulation class instead to get rid of the Singleton. Let me know what you think.